lib/client.js

```diff
@@ -78,6 +81,11 @@ module.exports = function (dependencies) {
               this.config.token.regenerate(tokenGeneration);
               resolve(this.write(notification, device, retryCount + 1));
               return;
+            } else if (status === "500" && response.reason === "InternalServerError") {
+              stream.connection.close();
+              let error = new VError("Error 500, stream ended unexpectedly");
```

Comment:
I too feel the same, the library should allow the users to handle error on their end instead of stream error.